titiler/openeo/stacapi.py

```diff
@@ -220,6 +221,115 @@ class LoadCollection:
 
     stac_api: stacApiBackend = field()
 
+    def _estimate_output_dimensions(
+        self,
+        items: List[Dict],
+        spatial_extent: Optional[BoundingBox],
+        width: Optional[int] = None,
+        height: Optional[int] = None,
+    ) -> Dict[str, Any]:
+        """
+        Estimate output dimensions based on items and spatial extent.
+
+        Args:
+            items: List of STAC items
+            spatial_extent: Bounding box for the output
+            width: Optional user-specified width
+            height: Optional user-specified height
+
+        Returns:
+            Dictionary containing:
+                - width: Estimated or specified width
+                - height: Estimated or specified height
+                - item_crs: CRS of the items
+                - crs: Target CRS to use
+                - bbox: Bounding box as a list [west, south, east, north]
+        """
+        # Extract CRS and resolution information from items
+        item_crs = None
+        x_resolutions = []
+        y_resolutions = []
+
+        for item in items:
+            with SimpleSTACReader(item) as src_dst:
+                if item_crs is None:
+                    item_crs = src_dst.crs
+                elif not item_crs.equals(src_dst.crs):
+                    raise ValueError(
+                        f"Mixed CRS in items: found {src_dst.crs} but expected {item_crs}"
+                    )
+
+                # Get resolution information
+                transform = src_dst.transform
+                x_resolutions.append(abs(transform.a))
+                y_resolutions.append(abs(transform.e))
```

Comment:
```suggestion
                x_resolutions.append(abs(src_dst.transform.a))
                y_resolutions.append(abs(src_dst.transform.e))
```
avoid creating variable